GoogleCloudPlatform/appengine-sidecars-docker
# Fix Time (#1)

* Added build-essential 

Added build-essential to list of apt-get install packages.

We need "make" to build native extensions.  Not sure how this worked before.  Maybe the base image used to include it.

* Update nginx proxy Dockerfile.

The update installs a newer build of NGINX with Endpoints support.

* Use gcr.io/google_appengine/base image.

The hyphen and underscore images are aliases of the same image
but using underscore for consistency.

* Expose port 24224 and listen on it

* Rewrite app container log tags according to stdout/stderr

* Update fluent-plugin-google-cloud to 0.5.2 to resolve build issue with transitive dependency

* Update the nginx Docker image.

1. Install cron so that log rotation can work.
2. Optionally accept Service Name and Service Version as parameters for
Endpoints Server Proxy.

* Address review comments

* memcache_proxy/main: fix flag handling, replace panic with Fatalf (#18)

* memcache_proxy: return "App Engine" for "version" (#19)

* testing: add tests and travis config (#21)

* Add regex to parse time from the $msec directive of nginx. (#24)

Add regex to parse time from the  directive of nginx.

* Revert "Expose an HTTP socket to receive app container logs" (#22)

* Add regex to parse time from the $request_time directive of nginx. (#25)

* Add latencySeconds parsing to the nginx log regex.

* Rename optional param for consistency with timestampSeconds.

* Error logging and tag renaming for nginx logs. (#26)

* Add the nginx error log configuration.

* Change tag naming based on product decision.

* Changes for consistency with other rules.

* Update fluentd config to include the log level, tid, and pid in the message (#27)

* Filter health check requests into a separate tag. (#29)

* Rename syslog -> vm.syslog as part of log tag cleanup. (#30)

* memcache_proxy: add version to STAT command (#28)

* memcache_proxy: use random open port for tests (#20)

* Tag all log entries with instance name label (#31)
